gulpfile.js

```diff
@@ -1,66 +1,61 @@
-var gulp = require("gulp");
-var concat = require("gulp-concat");
-var browserify = require("gulp-browserify");
+const browserify = require("browserify");
+const gulp = require("gulp");
+const log = require("gulplog");
+const rename = require("gulp-rename");
+const source = require("vinyl-source-stream");
+const buffer = require("vinyl-buffer");
+const globby = require("globby");
+const through = require("through2");
+
+const standalone = "HFX";
+const main = "./src/HFX.js";
+const build = [main, "./src/sections/*.js", "./src/features/**/*.js"];
+const watch = build.concat(["./src/core/*.js"]);
+const browserifyOptions = (entries) => Object({
+  entries: entries,
+  standalone: standalone
+});
 
 gulp.task("build", asyncComplete => {
-  gulp.src(["src/_core/HFX.js"])
-    .pipe(concat("HFX.js"))
-    .pipe(browserify())
-    .pipe(gulp.dest("./release/js/"));
-
-  gulp.src(["src/modules/global/**.js", "src/_core/modules/**.js"])
-    .pipe(concat("Global.js"))
-    .pipe(browserify())
-    .pipe(gulp.dest("./release/js/"));
-
-  gulp.src(["src/modules/game/**.js", "src/_core/modules/**.js"])
-    .pipe(concat("Game.js"))
-    .pipe(browserify())
-    .pipe(gulp.dest("./release/js/"));
-
-  gulp.src(["src/modules/threads/**.js", "src/_core/modules/**.js"])
-    .pipe(concat("Threads.js"))
-    .pipe(browserify())
-    .pipe(gulp.dest("./release/js/"));
-
-  gulp.src(["src/modules/pms/**.js", "src/_core/modules/**.js"])
-    .pipe(concat("PMs.js"))
-    .pipe(browserify())
-    .pipe(gulp.dest("./release/js/"));
+  const bundledStream = through();
+
+  bundledStream
```

Comment:
Can you explain bundledStream and globby?
Just an IO library?

Reply:
Globby is a package for finding files matching the glob expression. I believe `through` is just a wrapper for the normal transform stream that `gulp` would create, so that it can be passed in to `browserify`. The `gulp-browserify` package is no longer maintained and they link some recipes maintained by `gulp` to use. I was working off https://github.com/gulpjs/gulp/blob/master/docs/recipes/browserify-with-globs.md.